src/components/manager/DatasetSelectionModal.js

```diff
@@ -0,0 +1,43 @@
+import React, {useState} from "react";
+import PropTypes from "prop-types";
+
+import {Form, Modal} from "antd";
+
+import DatasetTreeSelect from "./DatasetTreeSelect";
+
+import {nop} from "../../utils/misc";
+
+const DatasetSelectionModal = ({dataType, title, visible, onCancel, onOk}) => {
+
+    const [selectedProject, setSelectedProject] = useState(undefined);
+    const [selectedDataset, setSelectedDataset] = useState(undefined);
+
+    const onChangeInner = (project, dataset) => {
```

Comment:
passed to a nested component, so this should be a `useCallback` with no dependencies